keys/keys.go

```diff
@@ -70,6 +70,42 @@ func MakeTableMetadataKey(namespaceID uint32, tableName string) proto.Key {
 	return k
 }
 
+// indexKeyWidth returns the width of all the columnValues forming the index key.
+func indexKeyWidth(columnValues ...[]byte) (width int) {
+	for _, value := range columnValues {
+		width += len(value)
+	}
+	return
+}
+
+// populateTableIndexKey populates the key passed in with the
+// order encoded values forming the index key.
+func populateTableIndexKey(key []byte, tableID, indexID uint32, columnValues ...[]byte) []byte {
+	key = append(key, TableDataPrefix...)
+	key = encoding.EncodeUvarint(key, uint64(tableID))
+	key = encoding.EncodeUvarint(key, uint64(indexID))
+	for _, value := range columnValues {
+		key = encoding.EncodeBytes(key, value)
+	}
+	return key
+}
+
+// MakeTableIndexKey returns a primary or a secondary index key.
+func MakeTableIndexKey(tableID, indexID uint32, columnValues ...[]byte) proto.Key {
+	k := make([]byte, 0, len(TableDataPrefix)+20+indexKeyWidth(columnValues...))
+	k = populateTableIndexKey(k, tableID, indexID, columnValues...)
+	return k
+}
+
+// MakeTableDataKey returns a key to a value at a specific row and column
+// in the table.
+func MakeTableDataKey(tableID, indexID, columnID uint32, columnValues ...[]byte) proto.Key {
+	k := make([]byte, 0, len(TableDataPrefix)+30+indexKeyWidth(columnValues...))
```

Comment:
enough space to hold the encoded ints

On Fri, Jun 12, 2015 at 12:22 PM Tamir Duberstein <notifications@github.com>
wrote:

> In keys/keys.go
> <https://github.com/cockroachdb/cockroach/pull/1356#discussion_r32330583>:
>
> > +		key = encoding.EncodeBytes(key, value)
> > +	}
> > +	return key
> > +}
> > +
> > +// MakeTableIndexKey returns a primary or a secondary index key.
> > +func MakeTableIndexKey(tableID, indexID uint32, columnValues ...[]byte) proto.Key {
> > +	k := make([]byte, 0, len(TableDataPrefix)+20+indexKeyWidth(columnValues...)
>
> )
> > +	k = populateTableIndexKey(k, tableID, indexID, columnValues...)
> > +	return k
> > +}
> > +
> > +// MakeTableDataKey returns a key to a value at a specific row and column
> > +// in the table.
> > +func MakeTableDataKey(tableID, indexID, columnID uint32, columnValues ...[]byte) proto.Key {
> > +	k := make([]byte, 0, len(TableDataPrefix)+30+indexKeyWidth(columnValues...))
>
> and this one
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/cockroachdb/cockroach/pull/1356/files#r32330583>.
>